module/VuFind/src/VuFind/Content/Covers/Orb.php

```diff
@@ -0,0 +1,129 @@
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Frédéric Demians <f.demians@tamil.fr>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Content\Covers;
+
+/**
+ * Orb cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Frédéric Demians <f.demians@tamil.fr>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:content_provider_components
+ */
+class Orb extends \VuFind\Content\AbstractCover
+    implements \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Base URL for Orb API
+     *
+     * @var string
+     */
+    protected $url;
+
+    /**
+     * API user for Orb
+     *
+     * @var string
+     */
+    protected $apiUser;
+
+    /**
+     * API key for Orb
+     *
+     * @var string
+     */
+    protected $apiKey;
+
+    /**
+     * Constructor
+     *
+     * @param string $url     Base URL for Orb
+     * @param string $apiUser API key for Orb
+     * @param string $apiKey  API key for Orb
+     */
+    public function __construct($url, $apiUser, $apiKey)
+    {
+        $this->url = $url;
+        $this->apiUser = $apiUser;
+        $this->apiKey = $apiKey;
+        $this->supportsIsbn = $this->cacheAllowed = true;
+    }
+
+    /**
+     * Get an HTTP client
+     *
+     * @param string $url URL for client to use
+     *
+     * @return \Laminas\Http\Client
+     */
+    protected function getHttpClient($url = null)
+    {
+        if (null === $this->httpService) {
+            throw new \Exception('HTTP service missing.');
+        }
+        return $this->httpService->createClient($url);
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        $ean = $ids['isbn']->get13();
+
+        $url = 'https://' . $this->apiUser . ':' . $this->apiKey . '@' .
+               $this->url . '/products?eans=' . $ean . '&sort=ean_asc';
+        $result = $this->getHttpClient($url)->send();
+        if ($result->isSuccess()) {
+            $data = $result->getBody();
+            $json = json_decode($data, true);
+            foreach ($json['data'] as $title) {
+                if ($title['ean13'] == $ean) {
+                    $size = $size == 'small' ? 'thumbnail' : 'original';
```

Comment:
`$size =` is doubled here for some reason.

Reply:
It's the ternary operator. If $size == 'small', transform $size into 'thumbnail', otherwise 'original'. Is there anything wrong syntactically ?

Reply:
After the re-assignation, there is a return. So there is not risk... The return data structure returned by the API looks like this:

```json
{
  "data": [
    {
      "ean13": "9782247197217",
      "images": {
        "front": {
          "original": {
            "src": "https://products-images.di-static.com/image/orb-ec3af334e23ff377fc14ad32c1497305/9782247197217-475x500-1.jpg"
          },
          "thumbnail": {
            "src": "https://products-images.di-static.com/image/orb-ec3af334e23ff377fc14ad32c1497305/9782247197217-120x160-1.jpg"
          }
        }
      }
    }
  ]
}
```

I will add a test to be sure that a URL is set.